core/src/main/java/org/infinispan/container/impl/DefaultSegmentedDataContainer.java

```diff
@@ -234,11 +234,11 @@ public void removeSegments(IntSet segments) {
 
    private void startNewMap(int segment) {
       if (maps.get(segment) == null) {
-         PeekableTouchableContainerMap<K, V> newMap = new PeekableTouchableContainerMap<>(mapSupplier.get());
+         PeekableTouchableMap<K, InternalCacheEntry<K, V>> newMap = mapSupplier.get();
          // Just in case of concurrent starts - this shouldn't be possible
-         if (!maps.compareAndSet(segment, null, newMap) && newMap.delegate() instanceof AutoCloseable) {
+         if (!maps.compareAndSet(segment, null, newMap) && newMap instanceof AutoCloseable) {
```

Comment:
https://github.com/infinispan/infinispan/blob/master/core/src/main/java/org/infinispan/container/offheap/OffHeapConcurrentMap.java#L84 is the one that doesn't.

Reply:
`PeekableTouchableContainerMap` doesn't implement `AutoCloseable` either

Reply:
Yeah it doesn't need it though, it is all on heap with nothing started :)

Reply:
Unless I am missing a reference somewhere, in which case it would need fixing.

Reply:
I'm not saying it needs `AutoCloseable`, I'm just saying this code is never executed because none of the classes implement `AutoCloseable`.

Reply:
OffHeapConcurrentMap does as I linked, I just accidentally put master instead of 9.4.x
https://github.com/infinispan/infinispan/blob/9.4.x/core/src/main/java/org/infinispan/container/offheap/OffHeapConcurrentMap.java#L33

Reply:
ah it used to, refactoring at its best.

Reply:
It does on master, but not on 9.4.x is the problem.

Reply:
LOL, I thought I was looking at the master code in my working dir... it took me a second to realize that I had created my task branch from 9.4.x instead of master!